apps/src/craft/code-connection/craft.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import {singleton as studioApp} from '../../StudioApp';
+import codegen from '../../codegen';
+import {Provider} from 'react-redux';
+import AppView from '../../templates/AppView';
+import CraftVisualizationColumn from './CraftVisualizationColumn';
+import {getStore} from '../../redux';
+
+const MEDIA_URL = '/blockly/media/craft/';
+
+/**
+ * Create a namespace for the application.
+ */
+const Craft = module.exports;
```

Comment:
this is an ancient and no longer necessary pattern.  You are better off doing:

```jsx
export default {
  init() {
    ...
  },

  reset() {
    ...
  },

  ...
};
```

Also, if this `Craft` object is going to have any state, then it should really be instantiable, so that testing it possible.